src/renderer/component/threeViewer/index.jsx

```diff
@@ -0,0 +1,288 @@
+    controls.maxDistance = 50;
+    controls.autoRotate = autoRotate;
+    return controls;
+  }
+
+  createGeometry(data) {
+    const geometry = new THREE.Geometry();
+    geometry.fromBufferGeometry(data);
+    geometry.computeBoundingBox();
+    geometry.computeVertexNormals();
+    geometry.center();
+    geometry.rotateX(-Math.PI / 2);
+    geometry.lookAt(new THREE.Vector3(0, 0, 1));
+    return geometry;
+  }
+
+  createWireFrame(group) {
+    const wireframeGeometry = new THREE.WireframeGeometry(group.geometry);
+    const wireframeMaterial = new THREE.LineBasicMaterial({
+      opacity: 0,
+      transparent: true,
+      linewidth: 1,
+    });
+    // Set material color
+    wireframeMaterial.color.set(this.materialColors.green);
+    this.wireframe = new THREE.LineSegments(wireframeGeometry, wireframeMaterial);
+    group.add(this.wireframe);
+  }
+
+  createMesh(geometry) {
+    const material = new THREE.MeshPhongMaterial({
+      opacity: 1,
+      transparent: true,
+      depthWrite: true,
+      vertexColors: THREE.FaceColors,
+      // Positive value pushes polygon further away
+      polygonOffsetFactor: 1,
+      polygonOffsetUnits: 1,
+    });
+
+    // Set material color
+    material.color.set(this.materialColors.green);
+
+    const mesh = new THREE.Mesh(geometry, material);
+
+    // Assign name
+    mesh.name = 'objectGroup';
+
+    this.scene.add(mesh);
+    this.fitMeshToCamera(mesh);
+    this.createWireFrame(mesh);
+    this.updateControlsTarget(mesh.position);
+    return mesh;
+  }
+
+  toggleWireFrame(show = false) {
+    this.wireframe.opacity = show ? 1 : 0;
+    this.mesh.material.opacity = show ? 0 : 1;
+  }
+
+  fitMeshToCamera(group) {
+    const max = { x: 0, y: 0, z: 0 };
+    const min = { x: 0, y: 0, z: 0 };
+
+    group.traverse(child => {
+      if (child instanceof THREE.Mesh) {
+        const box = new THREE.Box3().setFromObject(group);
+        // Max
+        max.x = box.max.x > max.x ? box.max.x : max.x;
+        max.y = box.max.y > max.y ? box.max.y : max.y;
+        max.z = box.max.z > max.z ? box.max.z : max.z;
+        // Min
+        min.x = box.min.x < min.x ? box.min.x : min.x;
+        min.y = box.min.y < min.y ? box.min.y : min.y;
+        min.z = box.min.z < min.z ? box.min.z : min.z;
+      }
+    });
+
+    const meshY = Math.abs(max.y - min.y);
+    const meshX = Math.abs(max.x - min.x);
+    const scaleFactor = 15 / Math.max(meshX, meshY);
+
+    group.scale.set(scaleFactor, scaleFactor, scaleFactor);
+    group.position.setY((meshY / 2) * scaleFactor);
+    group.position.multiplyScalar(-1);
+    group.position.setY((meshY * scaleFactor) / 2);
+  }
+
+  startLoader() {
+    const { source } = this.props;
+
+    if (source) {
+      ThreeLoader(source, this.renderModel.bind(this), {
+        onStart: this.handleStart(this),
+        onLoad: this.handleReady.bind(this),
+        onError: this.handleError.bind(this),
+        onProgress: this.handleProgress.bind(this),
```

Comment:
You shouldn't need to bind `this` in this functions. `handleStart` (and the others) uses the same `this` reference